RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

```diff
@@ -425,6 +436,40 @@ bool RecHitTools::isHalfCell(const DetId& id) const {
   return ishalf;
 }
 
+int RecHitTools::getLayerType(const DetId& id) const {
```

Comment:
This name is utterly confusing. There are layers with mixe types and, depending on the supplied `DetId`, the returned value will be different for the very same layer. In fact, AFAIC, this function is called in only one place to set an attribute of a cluster, which has a much local definition wrt `layer`.
Suggest renaming both the `enum` and this method.